src/Processors/Formats/Impl/ParallelFormattingOutputFormat.cpp

```diff
@@ -96,9 +97,9 @@ namespace DB
     }
 
 
-    void ParallelFormattingOutputFormat::finishAndWait() noexcept
+    void ParallelFormattingOutputFormat::finishAndWait(bool emergency_stop_) noexcept
     {
-        emergency_stop = true;
+        emergency_stop = emergency_stop_;
```

Comment:
What if `emergency_stop` is already set to `true`, but you set it to `false`? Let's do something like:
```
bool expected = false;
emergency_stop. compare_exchange_strong(expected, emergency_stop_);
```